src/cogent3/app/data_store.py

```diff
@@ -570,15 +570,15 @@ def _has_other_suffixes(self, path, suffix):
         p = Path(path)
         for f in p.iterdir():
             if get_format_suffixes(str(f))[0] != suffix:
-                return False
-        return True
+                return True
+        return False
 
     def _source_create_delete(self, if_exists, create):
         exists = os.path.exists(self.source)
         if exists and if_exists == RAISE:
             raise RuntimeError(f"'{self.source}' exists")
         elif exists and if_exists == OVERWRITE:
-            if self._has_other_suffixes(self.source, self.suffix):
+            if not self._has_other_suffixes(self.source, self.suffix):
```

Comment:
Oops, I think I should have asked for you to make this call the same as the one you do below, i.e. raise `RuntimeError` with the same message as below. So that means this does become `if self._has_other...: raise...`